SimMuon/GEMDigitizer/src/GEMBkgModel.cc

```diff
@@ -0,0 +1,123 @@
+      GE11ElecBkgParam0_(config.getParameter<double>("GE11ElecBkgParam0")),
+      GE11ElecBkgParam1_(config.getParameter<double>("GE11ElecBkgParam1")),
+      GE11ElecBkgParam2_(config.getParameter<double>("GE11ElecBkgParam2")),
+      GE21ElecBkgParam0_(config.getParameter<double>("GE21ElecBkgParam0")),
+      GE21ElecBkgParam1_(config.getParameter<double>("GE21ElecBkgParam1")),
+      GE21ElecBkgParam2_(config.getParameter<double>("GE21ElecBkgParam2")),
+      GE11ModNeuBkgParam0_(config.getParameter<double>("GE11ModNeuBkgParam0")),
+      GE11ModNeuBkgParam1_(config.getParameter<double>("GE11ModNeuBkgParam1")),
+      GE11ModNeuBkgParam2_(config.getParameter<double>("GE11ModNeuBkgParam2")),
+      GE21ModNeuBkgParam0_(config.getParameter<double>("GE11ModNeuBkgParam0")),
+      GE21ModNeuBkgParam1_(config.getParameter<double>("GE11ModNeuBkgParam1")),
+      GE21ModNeuBkgParam2_(config.getParameter<double>("GE11ModNeuBkgParam2")) {}
+
+GEMBkgModel::~GEMBkgModel() {}
+
+void GEMBkgModel::simulate(const GEMEtaPartition* roll,
+                           const edm::PSimHitContainer&,
+                           CLHEP::HepRandomEngine* engine,
+                           Strips& strips_,
+                           DetectorHitMap& detectorHitMap_) {
+  const GEMDetId& gemId(roll->id());
+  const int nstrips(roll->nstrips());
+  double trArea(0.0);
+  double trStripArea(0.0);
+  if (gemId.region() == 0) {
+    throw cms::Exception("Geometry")
+        << "GEMSynchronizer::simulate() - this GEM id is from barrel, which cannot happen.";
+  }
+  const TrapezoidalStripTopology* top_(dynamic_cast<const TrapezoidalStripTopology*>(&(roll->topology())));
+  const float striplength(top_->stripLength());
+  trStripArea = (roll->pitch()) * striplength;
+  trArea = trStripArea * nstrips;
+  const int nBxing(maxBunch_ - minBunch_ + 1);
+  const float rollRadius(
+      fixedRollRadius_
+          ? top_->radius()
+          : top_->radius() + CLHEP::RandFlat::shoot(engine, -1. * top_->stripLength() / 2., top_->stripLength() / 2.));
+
+  //calculate noise from model
+  double averageNeutralNoiseRatePerRoll = 0.;
+  double averageNoiseElectronRatePerRoll = 0.;
+  double averageNoiseRatePerRoll = 0.;
+  if (gemId.station() == 1) {
+    //simulate neutral background for GE1/1
+    averageNeutralNoiseRatePerRoll =
+        (GE11ModNeuBkgParam0_ + GE11ModNeuBkgParam1_ * rollRadius +
+         GE11ModNeuBkgParam2_ * rollRadius * rollRadius);  //simulate electron background for GE1/1
+    if (simulateElectronBkg_)
+      averageNoiseElectronRatePerRoll =
+          (GE11ElecBkgParam0_ + GE11ElecBkgParam1_ * rollRadius + GE11ElecBkgParam2_ * rollRadius * rollRadius);
+
+    // Scale up/down for desired instantaneous lumi (reference is 5E34, double from config is in units of 1E34)
+    averageNoiseRatePerRoll = averageNeutralNoiseRatePerRoll + averageNoiseElectronRatePerRoll;
+    averageNoiseRatePerRoll *= instLumi_ * rateFact_ * 1.0 / referenceInstLumi_;
```

Comment:
multiplication by 1.0 is pointless (unless 1.0 indicates some kind of unit conversion, in which case it should be a named constant)
apply also to the case a few lines below this